lib/src/components/NxDropdown/stateful/__tests__/NxStatefulDropdown.test.tsx

```diff
@@ -5,192 +5,311 @@
  * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
  */
 import React from 'react';
-import { act } from 'react-dom/test-utils';
-import { mount, ReactWrapper } from 'enzyme';
-import 'jest-enzyme';
-import * as enzymeUtils from '../../../../__testutils__/enzymeUtils';
 
-import NxButton from '../../../NxButton/NxButton';
-import NxDropdown from '../../NxDropdown';
+import { render, screen, fireEvent, within, createEvent } from '@testing-library/react';
+import { rtlRenderElement, rtlRender, userEvent } from '../../../../__testutils__/rtlUtils';
+
 import NxStatefulDropdown, { Props } from '../NxStatefulDropdown';
 
 describe('NxStatefulDropdown', () => {
-  let container: HTMLDivElement | null;
-
   const minimalProps = {
-        label: 'dropdown',
-        children: <a>Hello</a>
-      },
-      getShallowComponent = enzymeUtils.getShallowComponent<Props>(NxStatefulDropdown, minimalProps),
-      getMountedComponent = enzymeUtils.getMountedComponent<Props>(NxStatefulDropdown, minimalProps);
-
-  beforeEach(function() {
-    // Avoid rendering directly on the body.
-    container = document.createElement('div');
-    document.body.appendChild(container);
-  });
-
-  afterEach(function() {
-    if (container) {
-      document.body.removeChild(container);
-      container = null;
-    }
-  });
+    label: 'dropdown-toggle',
+    children: <a>Hello</a>
+  };
 
-  it('renders an NxDropdown component', function() {
-    const component = getShallowComponent();
-    expect(component).toMatchSelector(NxDropdown);
-    expect(component).toHaveProp('label', 'dropdown');
-  });
+  const renderEl = rtlRenderElement<Props>(NxStatefulDropdown, minimalProps);
+  const quickRender = rtlRender<Props>(NxStatefulDropdown, minimalProps);
 
-  it('renders an NxDropdown component passing on the given props', function() {
-    const component = getShallowComponent({
+  it('renders an NxStatefulDropdown toggle element correctly with the given props', function() {
+    const { container } = quickRender({
       variant: 'tertiary',
       className: 'extra-class',
-      disabled: true,
-      toggleTooltip: 'a tooltip'
+      disabled: true
     });
-    const statelessDropdown = component.find(NxDropdown);
 
-    expect(component).toMatchSelector(NxDropdown);
-    expect(statelessDropdown).toHaveProp('label', 'dropdown');
-    expect(statelessDropdown).toHaveProp('variant', 'tertiary');
-    expect(statelessDropdown).toHaveProp('disabled', true);
-    expect(statelessDropdown).toHaveProp('toggleTooltip', 'a tooltip');
-    expect(statelessDropdown).toHaveProp('children', (<a>Hello</a>));
+    const dropdown = container.querySelector('.nx-dropdown');
+    const toggleButton = screen.getByRole('button', { name: 'dropdown-toggle'});
+
+    expect(dropdown).toHaveClass('extra-class');
```

Comment:
Extra classes should be tested with this familiar pattern: https://github.com/sonatype/sonatype-react-shared-components/pull/909/files#diff-8db925aaf03e68533aaa4b6fa317b86cc22eb19cf52f14631de4c54e5851039bR36

Reply:
> Extra classes should be tested with this familiar pattern: https://github.com/sonatype/sonatype-react-shared-components/pull/909/files#diff-8db925aaf03e68533aaa4b6fa317b86cc22eb19cf52f14631de4c54e5851039bR36

Cleaned it up here: 041a235e97a4bd79499e548d3229ff2fbfaf70db